easybuild/easyblocks/s/scalapack.py

```diff
@@ -53,187 +53,205 @@
-            mpicc = os.getenv('MPICC')
-            mpif77 = os.getenv('MPIF77')
-            mpif90 = os.getenv('MPIF90')
-        elif self.toolchain.mpi_family() in known_mpi_libs:
-            mpicc = 'mpicc'
-            mpif77 = 'mpif77'
-            mpif90 = 'mpif90'
-        else:
-            raise EasyBuildError("Don't know which compiler commands to use.")
-
-        # determine build options BLAS and LAPACK libs
-        extra_makeopts = []
-
-        acml = get_software_root(Acml.LAPACK_MODULE_NAME[0])
-        lapack = get_software_root(Lapack.LAPACK_MODULE_NAME[0])
-        openblas = get_software_root(OpenBLAS.LAPACK_MODULE_NAME[0])
-        intelmkl = get_software_root(IntelMKL.LAPACK_MODULE_NAME[0])
-
-        if lapack:
-            extra_makeopts.append('LAPACKLIB=%s' % os.path.join(lapack, 'lib', 'liblapack.a'))
-
-            for blas in [Atlas, Blis, GotoBLAS]:
-                blas_root = get_software_root(blas.BLAS_MODULE_NAME[0])
-                if blas_root:
-                    blas_libs = ' '.join(['-l%s' % lib for lib in blas.BLAS_LIB])
-                    extra_makeopts.append('BLASLIB="-L%s %s -lpthread"' % (os.path.join(blas_root, 'lib'), blas_libs))
-                    break
-
-            if not blas_root:
-                raise EasyBuildError("Failed to find a known BLAS library, don't know how to define 'BLASLIB'")
-
-        elif acml:
-            acml_base_dir = os.getenv('ACML_BASEDIR', 'NO_ACML_BASEDIR')
-            acml_static_lib = os.path.join(acml, acml_base_dir, 'lib', 'libacml.a')
-            extra_makeopts.extend([
-                'BLASLIB="%s -lpthread"' % acml_static_lib,
-                'LAPACKLIB=%s' % acml_static_lib
-            ])
-        elif openblas:
-            libdir = os.path.join(openblas, 'lib')
-            blas_libs = ' '.join(['-l%s' % lib for lib in OpenBLAS.BLAS_LIB])
-            extra_makeopts.extend([
-                'BLASLIB="-L%s %s -lpthread"' % (libdir, blas_libs),
-                'LAPACKLIB="-L%s %s"' % (libdir, blas_libs),
-            ])
-        elif intelmkl:
-            libdir = os.path.join(intelmkl, 'mkl', 'lib', 'intel64')
-            blas_libs = os.environ['LIBLAPACK']
-            extra_makeopts.extend([
-                'BLASLIB="-L%s %s -lpthread"' % (libdir, blas_libs),
-                'LAPACKLIB="-L%s %s"' % (libdir, blas_libs),
-            ])
-        else:
-            raise EasyBuildError("Unknown LAPACK library used, no idea how to set BLASLIB/LAPACKLIB make options")
-
-        # build procedure changed in v2.0.0
-        if self.loosever < LooseVersion('2.0.0'):
-
-            blacs = get_software_root(Blacs.BLACS_MODULE_NAME[0])
-            if not blacs:
-                raise EasyBuildError("BLACS not available, yet required for ScaLAPACK version < 2.0.0")
-
-            # determine interface
-            interface = det_interface(self.log, os.path.join(blacs, 'bin'))
+        if not self.use_cmake:
```

Comment:
can you put this in a separate function please, it would reduce the nesting a bit.

e.g. `def build_step_cmake(self):` and `def_build_step_make(self)` or similar.